SampleApplications/SDK/Opc.Ua.Client/CoreClientUtils.cs

```diff
@@ -92,9 +111,24 @@ public static IList<string> DiscoverServers(ApplicationConfiguration configurati
         /// </summary>
         /// <param name="discoveryUrl">The discovery URL.</param>
         /// <param name="useSecurity">if set to <c>true</c> select an endpoint that uses security.</param>
-        /// <param name="operationTimeout">Optional. Operation timeout in milliseconds.</param>
         /// <returns>The best available endpoint.</returns>
-        public static EndpointDescription SelectEndpoint(string discoveryUrl, bool useSecurity, int operationTimeout = -1)
+        public static EndpointDescription SelectEndpoint(string discoveryUrl, bool useSecurity)
+        {
+            return SelectEndpoint(discoveryUrl, useSecurity, DefaultDiscoverTimeout);
+        }
+
+        /// <summary>
+        /// Finds the endpoint that best matches the current settings.
+        /// </summary>
+        /// <param name="discoveryUrl">The discovery URL.</param>
+        /// <param name="useSecurity">if set to <c>true</c> select an endpoint that uses security.</param>
+        /// <param name="discoverTimeout">Operation timeout in milliseconds.</param>
+        /// <returns>The best available endpoint.</returns>
+        public static EndpointDescription SelectEndpoint(
+            string discoveryUrl,
+            bool useSecurity,
+            int discoverTimeout
+            )
         {
             // needs to add the '/discovery' back onto non-UA TCP URLs.
             if (discoveryUrl.StartsWith(Utils.UriSchemeHttps))
```

Comment:
>if (discoveryUrl.StartsWith(Utils.UriSchemeHttps)) [](start = 12, length = 50)

[Question] Is adding the "/discovery" also neccessary in http case?

Reply:
could be, but we don't support http here...

---
In reply to: [432605023](https://github.com/OPCFoundation/UA-.NETStandard/pull/998#discussion_r432605023) [](ancestors = 432605023)